Plugins/GeoModel/include/Acts/Plugins/GeoModel/GeoModelToDetVol.hpp

```diff
@@ -0,0 +1,27 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+#pragma once
+
+#include "Acts/Detector/DetectorVolume.hpp"
+
+class GeoShape;
+
+namespace Acts {
+
+namespace GeoModel {
+/// @brief Convert a GeoModel shape to a DetectorVolume
+///
+/// @param shape the GeoModel shape
+/// @param transform the transform to be applied
+/// @return the DetectorVolume
+void convertVolume(
+    const GeometryContext& context, const GeoShape& shape,
+    const std::string& name, const Transform3& transform,
+    std::vector<std::shared_ptr<Experimental::DetectorVolume>>& volumes);
```

Comment:
you could make this one also to be the return type of the method. Or does it append?

Reply:
Ok, yeah so this just appends to some vector you pass into it because `GeoShapeUnion` does not seem to transform correctly when you take the extent of it. This then led me into breaking the union up into multiple volumes so the easiest way to deal with it is just appending the volumes directly to some container.